lxd/storage/drivers/driver_dir.go

```diff
@@ -0,0 +1,864 @@
+package drivers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/gorilla/websocket"
+	"golang.org/x/sys/unix"
+
+	"github.com/lxc/lxd/lxd/migration"
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/lxd/rsync"
+	"github.com/lxc/lxd/lxd/storage/quota"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/units"
+)
+
+type dir struct {
+	common
+}
+
+func (d *dir) Name() string {
+	return "dir"
+}
+
+func (d *dir) Version() string {
+	return "1"
+}
+
+// OptimizedImages returns whether the storage driver will create Instances from snapshots of
+// images rather than unpack the image every time.
+func (d *dir) OptimizedImages() bool {
+	return false
+}
+
+// MigrationType returns the type of transfer method used when doing migrations between pools
+// of the same type.
+func (d *dir) MigrationTypes(contentType ContentType) []migration.Type {
+	if contentType != ContentTypeFS {
+		return nil
+	}
+
+	return []migration.Type{
+		{
```

Comment:
Probably ought to at least be moved to something in utils.go so we can include the same migration.Type for all those using rsync.

Reply:
@stgraber I'll move this function into the `common` driver, so for the drivers that use only rsync (and eventually dd for blocks) they won't need to implement this function at all.

For those that have their own native types as a preference first, they can use the following implementation:

```go
func (d *zfs) MigrationTypes(contentType ContentType) []migration.Type {
	myTypes := []migration.Type{} // My specific ZFS native type here.

        // Add the common rsync/dd types if needed here.
	myTypes = append(myTypes, d.common.MigrationTypes(contentType)...)
	return myTypes
}
```